lib/cli.js

```diff
@@ -99,6 +99,10 @@ exports.run = () => {
 		throw new Error(colors.error(figures.cross) + ' The TAP reporter is not available when using watch mode.');
 	}
 
+	if ((hasFlag('--watch') || hasFlag('-w')) && isCi) {
+		throw new Error(colors.error(figures.cross) + ' AVA will not run with the --watch (-w) option in CI, because CI processes should terminate, and with the --watch option, AVA will never terminate.');
```

Comment:
The explanation seems a bit long for a CLI error. Contrast with the TAP reporter where we just say "sorry, not available".